modules/mapbox/src/deck-utils.js

```diff
@@ -56,21 +52,35 @@ export function updateLayer(deck, layer) {
 }
 
 export function drawLayer(deck, layer) {
-  deck.props.userData.layerFilter = layer.id;
+  // set layerFilter to only allow the current layer
+  deck.layerManager.layerFilter = params => shouldDrawLayer(layer.id, params.layer);
   deck._drawLayers('mapbox-repaint');
-  deck.needsRedraw({clearRedrawFlags: true});
 }
 
-function filterLayer(deck, layer) {
-  const {layerFilter} = deck.props.userData;
-
-  if (typeof layerFilter !== 'string') {
-    return layerFilter;
+function afterRender(deck) {
+  const {mapboxLayers, isExternal} = deck.props.userData;
+
+  if (isExternal) {
+    // Draw non-Mapbox layers
+    const mapboxLayerIds = Array.from(mapboxLayers).map(layer => layer.id);
```

Comment:
Minor, but `Array.from` can take a second mapper argument, no need for the `map`

```suggestion
    const mapboxLayerIds = Array.from(mapboxLayers, layer => layer.id);
```